weatherman.py

```diff
@@ -0,0 +1,384 @@
+
+
+class Colors:
+    """Class for colored text on console 
+    """
+
+    BLUE = '\033[94m'
+    RED = '\033[91m'
+    ENDC = '\033[0m'
+
+
+def month_chart(year, month):
+    """Prints color charts on console of a given month
+    """
+
+    high_temps = calculations.get_attr_values(
+        weather_dataset, "Max TemperatureC",  year, month)
+
+    low_temps = calculations.get_attr_values(
+        weather_dataset, "Min TemperatureC", year, month)
+    
+    month_name = high_temps[0][1].strftime("%B")
+
+    # Saving max temperature and min temperature in dictionaries
+    # key is datetime object
+    high_temps = {attr[1]: attr[0] for attr in high_temps}
+    low_temps = {attr[1]: attr[0] for attr in low_temps}
+
+    # prints colored chart
+    print("{0} {1}".format(month_name, year))
+    for k in high_temps:
+        # low temp chart
+        print(k.strftime("%d"), end=" ")
+        for _ in range(low_temps[k]):
+            print(Colors.BLUE + "+" + Colors.ENDC, end="")
+        # high temp chart
+        for _ in range(high_temps[k]):
+            print(Colors.RED + "+" + Colors.ENDC, end="")
+
+        print("", str(low_temps[k])+"C - ", end="")
+        print(str(high_temps[k])+"C", end='\n')
+
+
+def string_to_date(date_string):
+    """Converts the string date, in format yyyy-mm-dd, to datetime   
+    """
+
+    try:
+        date = dt.datetime.strptime(date_string, "%Y-%m-%d")
+    except:
+        print("Invalid format of date found in dataset")
+    else:
+        return date
+
+
+def print_correct_format():
+    """Prints examples of correct command format and exits code
+
+    Helper Function that assists, when user enters invalid command
+    to execute code
+    """
+
+    print("Following are the correct command examples:\n")
+    print("python3 weatherman.py /path-to-zipfile -e 2011 ")
+    print("python3 weatherman.py /path-to-zipfile -e 2016 -a 2007/6" +
+          " -c 2009/5\n")
+    exit()
+
+
+def date_exists(year, month=-1):
+    """Returns True if given month/year exists in weather dataset
+
+    Args:
+        month (int, optional)
+        year (int) 
+
+    Returns:
+        [Boolean]
+    """
+
+    if year in weather_dataset:
+        if month != -1:
+            if month in weather_dataset[year]:
+                return True
+            else:
+                return False
+        return True
+    else:
+        return False
+
+
+if __name__ == "__main__":
+
+    argv = handle_sys_argv(sys.argv)
+    zip_path = argv["path"].strip()
+    filenames = extract_files(zip_path)
+    files_path = "./weatherfiles"
+
+    for file_name in filenames:
+        if file_name.endswith(".txt"):
+            year, month, month_data = handle_csv(files_path, file_name, ",")
+
+        elif file_name.endswith(".tsv"):
+            year, month, month_data = handle_csv(files_path, file_name, "\t")
+
+        elif file_name.endswith(".xlsx"):
+            year, month, month_data = handle_xlsx(files_path, file_name)
```

Comment:
@ahmadx16 It would be a cleaner approach if you can wrap all this conditional data reading to a simple function where you just call that and it will read the data based on the file type.
Something like this instead of these conditional line, these conditions can be moved into `get_month_data` function
```suggestion
       year, month, month_data = get_month_data(files_path, file_name)
```